### PR TITLE
feat: 🎸 exit code 2 if extra keys found

### PR DESCRIPTION
**Pull request by JerryDoubleU**
Closes: #34

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/transloco/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Change in the Keys Detective (find) behavior.
If there were some missing keys and the user would have the 'addMissingKey' option set to false, there would be an exit code 1 returned.

Issue Number: #34 

## What is the new behavior?

Extra behavior has been added. If there would be any missing key found, the script will end with an exit code 2.

## Does this PR introduce a breaking change?

```
[x] Yes
[ ] No
```

There are two scenarios which are affected.
1. The user sets the 'addMissingKeys' to true and there were some missing and extra keys found. Previously, the script would show 'Added all missing keys' message and exit with code 0.
Now, the script would also show this message, but due to having the extra keys would exit with code 2.
2. There are no missing keys, but there are some extra keys.
Previously, the script would exit with code 0.
Now, the script would exit with code 2.

## Other information

I am new to contributing to a project like this, so please let me know if the commit, branch, etc. are names correctly. Let me know if and how can I improve them. BTW I love Transloco! :)

### CODE REVIEW

**Comment by shaharkazaz** [comment]
@JerryDoubleU Thanks for the PR! but I wouldn't want to make this a breaking change, how about we make exiting with status code optional? so you could define in which cases the process should fail.

**Comment by JerryDoubleU** [comment]
@shaharkazaz I've adjusted the code. By default, the Keys Detective will behave as it did before. If the user adds a parameter `--exit-on-extra-keys` or uses a config setting `exitOnExtraKeys: true`, the app will exit with code 2 if some extra keys were detected.
Please have a look if that's what was on your mind.

**Comment by shaharkazaz** [comment]
@JerryDoubleU looks good! can you please rename to `emitErrorOnExtraKeys` 😄

**Comment by JerryDoubleU** [comment]
@shaharkazaz Sure! I've renamed the setting and the parameter. :heavy_check_mark:

File: README.md [hunk 1]
```diff
@@ -360,6 +360,13 @@ transloco-keys-manager find --add-missing-keys
 transloco-keys-manager find -a
 ```
 
+- `emitErrorOnExtraKeys`: Return exit code if extra keys were found by the detective (defaults to `false`)
```
Comment (shaharkazaz) [inline README.md:363]: Emit an error and exit the process if extra keys were found (defaults to `false`)

Comment (JerryDoubleU) [reply README.md]: Done

**Pull request merged**